src/darwin/Framework/CHIP/MTRControllerFactory.mm

```diff
@@ -274,6 +282,22 @@ - (BOOL)startup:(MTRControllerFactoryParams *)startupParams
             return;
         }
 
+        auto systemState = _controllerFactory->GetSystemState();
+        if (systemState == nullptr) {
+            MTR_LOG_ERROR("Error: %@", kErrorOtaProviderInit);
+            return;
+        }
+
+        _otaProviderDelegateBridge = new MTROTAProviderDelegateBridge(systemState->SystemLayer(), systemState->ExchangeMgr());
```

Comment:
This is not going to work right.  If we shut down the last controller (but not the factory), this exchange manager will be destroyed.

We need to either keep the SystemState alive via `RetainSystemState` on the factory (and then deal with it all continuing to run when all controllers are shut down) or we need to clear out this bridge object when our controller count goes to 0 and recreate/reinit it when we get a new controller brought up or something.  Basically at the places where we StartEventLoopTask/StopEventLoopTask in this file.

Reply:
I did not realised that the lifetime of `MTRControllerFactory` was different than the lifetime of the underlying `CHIPDeviceControllerFactory`. Thanks !

Reply:
The lifetimes of the _factories_ are the same.  The lifetime of the system state is not the same.